### DISCUSSION TITLE
History folder

### DISCUSSION THREAD

**Discussion by WS64** [koreader/koreader#7090] [Q&A]
How can I tell Koreader to write all the book metadata to koreade\history folder and not to a foobar.sdr folder where the books are?

**Comment by pazos** [comment]
You can't. The history folder is only available for **read only** storage. On writable storage the metadata is kept in a sidecar folder, as it provides some benefits (ie: the ability to move files to other paths even with an external program while keeping the metadata intact)

The only available option **right now** is hacking the code or place books in ro storage, but that's not really an option for most end users.

**Comment by poire-z** [comment]
See https://github.com/koreader/koreader/issues/4831#issuecomment-475885999 or https://github.com/koreader/koreader/issues/4831#issuecomment-548245804 : the idea is that you just need to hack the function `DocSettings.ensureSidecar()` and make it a no-op.
We could even have a setting (hidden or not) so that when set, the existing `DocSettings.ensureSidecar()` returns early.
It's just that this would be another code path/behaviour to test with/maintain, which might be a burden (I'm not sure the above hack still works :)

(I really don't like these discussions - I don't see them in my [main watch of KOReader](https://github.com/koreader/koreader/issues?q=sort:updated-desc) - there's no number indicator on the tab that could indicate there is new ones, or how many are unanswered - so it could take days before I go look at that section - And they are not returned in the main search, so it's a bit useless putting some info here...)
(edit: and they are not even references in the issues we link from here...)

**Reply by pazos** [reply]
I just got notified by your answer :p. 

Agree it could be more useful, but it can still help in the long run to have a base of questions, that could avoid further issue duplication.

**Comment by WS64** [comment]
Actually I solve it right now by mounting the partition (btw, that's the new Fat32 partition from the other thread!) readonly.
But I really wonder why this is not optional.
My workflow is to synchronize my books with a master folder on my harddrive. I would lose all the data with the next sync (well, okay, I could tell robocopy to ignore lua files).
From time to time I check all the history files for my highlights since I want to correct some mistakes in the books with them. When they are in different folders I have to search for them, much more convenient to have them all in one place.
I can live with the readonly solution, but I suggest to make this customizable!

**Reply by poire-z** [reply]
Well, just try what my old links suggested (I guess, you may already been modifying koreader code):
```diff
--- a/frontend/docsettings.lua
+++ b/frontend/docsettings.lua
 function DocSettings:ensureSidecar(sidecar)
+    if true then return end
     if lfs.attributes(sidecar, "mode") ~= "directory" then
        lfs.mkdir(sidecar)
     end
 end
```
and see how that works.
There might be other issues to solve, but with luck not many.
If it's not too hellish, we'll just change that to
```diff
-if true then return end
+if G_reader_settings:isTrue("no_sidecar_thank_you") then return end
```

**Comment by WS64** [comment]
@poire-z, I will for sure test, has to wait a bit since I need to open the reader again to switch to my test SD card.
But actually for the original person asking for that did alread agree that it worked.
I am not sure if I miss something in the filebrower (file is opened), since I probably never saw it anyway (I did something like that 6 years ago already and never changed the Koreader version after that, but I can't remember what I did!)

**Comment by farvardin** [comment]
I've sure I've seen in newest koreader that you can store the .sdr metadata to a single place, instead of having them in the .epub folders. 
I've looked at the koreader options 3 times, and I can't find it again. How can we achieve that?

**Reply by benoit-pierre** [reply]
In `Document ▸ Boot metadata location: …`.

**Reply by farvardin** [reply]
thank you. I couldn't find it, my koreader version on this computer was too old! (I had seen the option first from my ebook reader) I've upgraded it and it works now, that's great. I'll keep the .sdr inside book folders on my pocketbook, it's easier to manage this way, but on my computer I only use Koreader for quickly previewing epub.
